Sources/ClientRuntime/Networking/HashFunction.swift

```diff
@@ -69,6 +91,43 @@ enum HashFunction {
     }
 }
 
+extension HashFunction: Comparable {
+    /*
+     * Priority-order for validating checksum = [ CRC32C, CRC32, SHA1, SHA256 ]
+     * Order is determined by speed of the algorithm's implementation
+     * MD5 is not supported by list ordering
+     */
+    public static func < (lhs: HashFunction, rhs: HashFunction) -> Bool {
+        let order: [HashFunction] = [.crc32c, .crc32, .sha1, .sha256]
+
+        let lhsIndex = order.firstIndex(of: lhs) ?? Int.max
+        let rhsIndex = order.firstIndex(of: rhs) ?? Int.max
+
+        return lhsIndex < rhsIndex
+    }
+}
+
+extension [HashFunction] {
+    func getPriorityOrderValidationList() -> [HashFunction] {
+        // Filter out .md5 if present and then sort the remaining hash functions
+        return self.filter { $0 != .md5 }.sorted()
+    }
+}
+
+extension UInt32 {
+    func toBase64EncodedString() -> String {
+        // Create a Data instance from the UInt32 value
+        var value = self
+        let data = Data(bytes: &value, count: MemoryLayout<UInt32>.size)
+
+        // Reverse bytes if on a little-endian architecture
+        let endianAdjustedData = Data(data.reversed())
```

Comment:
How does this behave differently if on a big- or little-endian system?  From what I can tell, `reversed()` does the same on either.

Reply:
I think you want to use `UInt32.littleEndian` or `UInt32.bigEndian` to get the byte order you want, no matter the platform.

Reply:
Ah great point, Ill change this

Reply:
From what I gather, Apple platforms (Apple Silicon & x86_64) are little-endian.  But when we start talking about proprietary ARM on Linux, it could be either... IDK

Reply:
forcing one or the other using littleEndian vs bigEndian regardless seems to change the base64encoded output causing the request to fail against S3's servers. A solution I thought of was to reverse only if the system is littleEndian:

```
// Reverse bytes if on a little-endian architecture
if self == self.littleEndian {
    data = Data(data.reversed())
}
```

Reply:
Calling `.littleEndian` should have no effect on the value on a little-endian system. Vice-versa for calling `.bigEndian` on a big-endian system.

Assuming you want the bytes in big-endian order:
```swift
let value: UInt32 = 123456  // or whatever value
let bigEndianValue = value.bigEndian
let data = Data(bytes: &bigEndianValue, count: MemoryLayout<UInt32>.size) 
```
`data` should be your integer's 4 bytes in big-endian order.